Sources/WebService/AuthenticatedWebService/AuthenticatedWebService.swift

```diff
@@ -56,84 +56,83 @@ open class AuthenticatedWebService: WebService {
     self.configuration = configuration
     super.init(urlSession: urlSession)
   }
-  
+
   override public func execute<T>(urlRequest: URLRequest) -> AnyPublisher<T, Error> where T : Decodable {
     var urlRequest = urlRequest
-    var currentAccessToken: String?
 
-    authenticationQueue.sync {
-      currentAccessToken = self.tokenProvider.accessToken.value
+    func appendTokenAndExecute(accessToken: AccessToken) -> AnyPublisher<T, Error> {
+      urlRequest.setValue(self.configuration.authorizationHeaderScheme.rawValue + accessToken, forHTTPHeaderField: "Authorization")
```

Comment:
should be deffered

Reply:
again, the consumer of this is deferred already. There is no direct access to this publisher without any defer mechanism.